ArduCopter/commands.cpp

```diff
@@ -23,8 +23,8 @@ void Copter::update_home_from_EKF()
 void Copter::set_home_to_current_location_inflight() {
     // get current location from EKF
     Location temp_loc;
-    if (inertial_nav.get_location(temp_loc)) {
-        const struct Location &ekf_origin = inertial_nav.get_origin();
+    Location ekf_origin;
+    if (ahrs.get_location(temp_loc) && ahrs.get_origin(ekf_origin)) {
```

Comment:
This is safe, because the `inertial_nav.get_location()` wraps the AHRS call. With the EKF this will always fail unless origin is available, which means pulling the `ahrs.get_origin()` call up into this line represents on change in behavior.

This same logic is also used within Sub.